From 6476f2e17253e222ca78db16342b6f74d754864a Mon Sep 17 00:00:00 2001
From: Yiming Cao <cyiming@codeaurora.org>
Date: Tue, 10 May 2016 16:46:45 +0800
Subject: [PATCH] mm-video-v4l2:vdec: strip dependence on android

1, disable dynamic buffer mode
2, replace atomic operations with glib atomic operations
3, use "OMX_DEBUG_LEVEL" environment to define debug level
and upgrade the log by adding pid/tid/filename/line info

Change-Id: If2dd16656576b7616319c83e165316892d3c96d6
Signed-off-by: Yiming Cao <cyiming@codeaurora.org>
---
 .../vidc/common/inc/vidc_color_converter.h         |  4 +-
 mm-video-v4l2/vidc/common/inc/vidc_debug.h         | 31 ++++++--
 mm-video-v4l2/vidc/vdec/inc/hevc_utils.h           |  4 +-
 mm-video-v4l2/vidc/vdec/inc/omx_vdec.h             | 12 ++-
 mm-video-v4l2/vidc/vdec/inc/ts_parser.h            |  3 +-
 mm-video-v4l2/vidc/vdec/src/h264_utils.cpp         | 13 +++-
 mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp      | 90 +++++++++++++++++++++-
 7 files changed, 138 insertions(+), 19 deletions(-)

diff --git a/mm-video-v4l2/vidc/common/inc/vidc_color_converter.h b/mm-video-v4l2/vidc/common/inc/vidc_color_converter.h
index 5361c11..01e4552 100644
--- a/mm-video-v4l2/vidc/common/inc/vidc_color_converter.h
+++ b/mm-video-v4l2/vidc/common/inc/vidc_color_converter.h
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2012-2013, The Linux Foundation. All rights reserved.
+Copyright (c) 2012-2013, 2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are
@@ -29,7 +29,9 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <dlfcn.h>
 #include "C2DColorConverter.h"
 
+#ifdef _ANDROID_
 using namespace android;
+#endif
 class omx_c2d_conv
 {
     public:
diff --git a/mm-video-v4l2/vidc/common/inc/vidc_debug.h b/mm-video-v4l2/vidc/common/inc/vidc_debug.h
index f01a40e..ec8f390 100644
--- a/mm-video-v4l2/vidc/common/inc/vidc_debug.h
+++ b/mm-video-v4l2/vidc/common/inc/vidc_debug.h
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2013, The Linux Foundation. All rights reserved.
+Copyright (c) 2013, 2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
@@ -29,9 +29,6 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef __VIDC_DEBUG_H__
 #define __VIDC_DEBUG_H__
 
-#ifdef _ANDROID_
-#include <cstdio>
-
 enum {
    PRIO_ERROR=0x1,
    PRIO_INFO=0x1,
@@ -41,6 +38,9 @@ enum {
 
 extern int debug_level;
 
+#ifdef _ANDROID_
+#include <cstdio>
+
 #undef DEBUG_PRINT_ERROR
 #define DEBUG_PRINT_ERROR(fmt, args...) ({ \
       if (debug_level & PRIO_ERROR) \
@@ -62,10 +62,25 @@ extern int debug_level;
           ALOGD(fmt,##args); \
       })
 #else
-#define DEBUG_PRINT_ERROR printf
-#define DEBUG_PRINT_INFO printf
-#define DEBUG_PRINT_LOW printf
-#define DEBUG_PRINT_HIGH printf
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <string.h>
+#include <stdlib.h>
+#define gettid() syscall(SYS_gettid)
+#define getpid() syscall(SYS_getpid)
+#define __FILENAME__ (strrchr(__FILE__, '/') ? strrchr(__FILE__, '/') + 1 : __FILE__)
+#define DEBUG_PRINT_CTL(level, fmt, args...) \
+  do {                                       \
+    if ( level & debug_level )     \
+      printf("[%ld:%ld]:[%s:%d] " fmt" \n", getpid(), \
+        gettid(), __FILENAME__, __LINE__, ##args); \
+  } while (0)
+
+#define DEBUG_PRINT_ERROR(fmt, args...) DEBUG_PRINT_CTL(PRIO_ERROR, fmt, ##args)
+#define DEBUG_PRINT_INFO(fmt, args...)  DEBUG_PRINT_CTL(PRIO_INFO, fmt, ##args)
+#define DEBUG_PRINT_LOW(fmt, args...)  DEBUG_PRINT_CTL(PRIO_LOW, fmt, ##args)
+#define DEBUG_PRINT_HIGH(fmt, args...) DEBUG_PRINT_CTL(PRIO_HIGH, fmt, ##args)
 #endif
 
 #endif
diff --git a/mm-video-v4l2/vidc/vdec/inc/hevc_utils.h b/mm-video-v4l2/vidc/vdec/inc/hevc_utils.h
index 797d1d2..3bb68ef 100644
--- a/mm-video-v4l2/vidc/vdec/inc/hevc_utils.h
+++ b/mm-video-v4l2/vidc/vdec/inc/hevc_utils.h
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2013, The Linux Foundation. All rights reserved.
+Copyright (c) 2013, 2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
@@ -46,7 +46,9 @@ This module contains H264 video decoder utilities and helper routines.
 
 ========================================================================== */
 #include <stdio.h>
+#ifdef _ANDROID_
 #include <utils/Log.h>
+#endif
 #include "Map.h"
 #include "qtypes.h"
 #include "OMX_Core.h"
diff --git a/mm-video-v4l2/vidc/vdec/inc/omx_vdec.h b/mm-video-v4l2/vidc/vdec/inc/omx_vdec.h
index 18bafe2..cb580b6 100644
--- a/mm-video-v4l2/vidc/vdec/inc/omx_vdec.h
+++ b/mm-video-v4l2/vidc/vdec/inc/omx_vdec.h
@@ -47,7 +47,9 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <string.h>
 #include <inttypes.h>
 #include <cstddef>
+#ifdef _ANDROID
 #include <cutils/atomic.h>
+#endif
 
 static ptrdiff_t x;
 
@@ -70,11 +72,14 @@ static ptrdiff_t x;
 extern "C" {
 #include <utils/Log.h>
 }
+#endif // _ANDROID_
+#ifdef USE_ION
+#include <linux/msm_ion.h>
+#endif
 #include <linux/videodev2.h>
 #include <poll.h>
 #include "hevc_utils.h"
 #define TIMEOUT 5000
-#endif // _ANDROID_
 
 #if defined (_ANDROID_HONEYCOMB_) || defined (_ANDROID_ICS_)
 #include <media/hardware/HardwareAPI.h>
@@ -104,12 +109,15 @@ extern "C" {
 #endif
 #include "extra_data_handler.h"
 #include "ts_parser.h"
+#ifdef _COLOR_CONVERTION_
 #include "vidc_color_converter.h"
+#endif
 #include "vidc_debug.h"
 #ifdef _ANDROID_
 #include <cutils/properties.h>
 #else
 #define PROPERTY_VALUE_MAX 92
+#include <dlfcn.h>
 #endif
 extern "C" {
     OMX_API void * get_omx_component_factory_fn(void);
@@ -180,9 +188,7 @@ class VideoHeap : public MemoryHeapBase
 
 #define DESC_BUFFER_SIZE (8192 * 16)
 
-#ifdef _ANDROID_
 #define MAX_NUM_INPUT_OUTPUT_BUFFERS 64
-#endif
 
 #define OMX_FRAMEINFO_EXTRADATA 0x00010000
 #define OMX_INTERLACE_EXTRADATA 0x00020000
diff --git a/mm-video-v4l2/vidc/vdec/inc/ts_parser.h b/mm-video-v4l2/vidc/vdec/inc/ts_parser.h
index 2d5d1a4..baa7358 100644
--- a/mm-video-v4l2/vidc/vdec/inc/ts_parser.h
+++ b/mm-video-v4l2/vidc/vdec/inc/ts_parser.h
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2010-2015, The Linux Foundation. All rights reserved.
+Copyright (c) 2010-2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
@@ -42,6 +42,7 @@ extern "C" {
 #include<utils/Log.h>
 }
 #else
+#include <pthread.h>
 #define ALOGE(fmt, args...) fprintf(stderr, fmt, ##args)
 #endif /* _ANDROID_ */
 
diff --git a/mm-video-v4l2/vidc/vdec/src/h264_utils.cpp b/mm-video-v4l2/vidc/vdec/src/h264_utils.cpp
index 13d4eac..34b7713 100644
--- a/mm-video-v4l2/vidc/vdec/src/h264_utils.cpp
+++ b/mm-video-v4l2/vidc/vdec/src/h264_utils.cpp
@@ -1,5 +1,5 @@
 /*--------------------------------------------------------------------------
-Copyright (c) 2010 - 2013, The Linux Foundation. All rights reserved.
+Copyright (c) 2010 - 2013, 2016, The Linux Foundation. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
@@ -56,7 +56,18 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 extern "C" {
 #include<utils/Log.h>
 }
+#else
+#include "vidc_debug.h"
+
+#undef ALOGD
+#undef ALOGI
+#undef ALOGV
+#undef ALOGE
 
+#define ALOGD DEBUG_PRINT_HIGH
+#define ALOGI DEBUG_PRINT_INFO
+#define ALOGV DEBUG_PRINT_LOW
+#define ALOGE DEBUG_PRINT_ERROR
 #endif
 
 /* =======================================================================
diff --git a/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp b/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
index d996e0f..fa1f553 100644
--- a/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
+++ b/mm-video-v4l2/vidc/vdec/src/omx_vdec_v4l2.cpp
@@ -51,22 +51,25 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <fcntl.h>
 #include <limits.h>
 #include <stdlib.h>
+#ifdef _ANDROID_
 #include <media/hardware/HardwareAPI.h>
+#endif
 #include <media/msm_media_info.h>
 #include <sys/eventfd.h>
 
 #ifndef _ANDROID_
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <glib.h>
+#define strlcpy g_strlcpy
 #endif //_ANDROID_
 
 #ifdef _ANDROID_
 #include <cutils/properties.h>
+#include <qdMetaData.h>
 #undef USE_EGL_IMAGE_GPU
 #endif
 
-#include <qdMetaData.h>
-
 #ifdef ANDROID_JELLYBEAN_MR2
 #include "QComOMXMetadata.h"
 #endif
@@ -80,6 +83,10 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define BUFFER_LOG_LOC "/data/misc/media"
 
+#ifndef USE_ION
+#include "linux/android_pmem.h"
+#endif
+
 #ifdef OUTPUT_EXTRADATA_LOG
 FILE *outputExtradataFile;
 char output_extradata_filename [] = "/data/misc/media/extradata";
@@ -633,8 +640,8 @@ omx_vdec::omx_vdec(): m_error_propogated(false),
     in_reconfig(false),
     m_display_id(NULL),
     client_extradata(0),
-    m_reject_avc_1080p_mp (0),
 #ifdef _ANDROID_
+    m_reject_avc_1080p_mp (0),
     m_enable_android_native_buffers(OMX_FALSE),
     m_use_android_native_buffers(OMX_FALSE),
 #endif
@@ -653,7 +660,9 @@ omx_vdec::omx_vdec(): m_error_propogated(false),
     m_pipe_out = -1;
     m_poll_efd = -1;
     drv_ctx.video_driver_fd = -1;
+#ifdef USE_ION
     drv_ctx.extradata_info.ion.fd_ion_data.fd = -1;
+#endif
     /* Assumption is that , to begin with , we have all the frames with decoder */
     DEBUG_PRINT_HIGH("In %u bit OMX vdec Constructor", (unsigned int)sizeof(long) * 8);
     memset(&m_debug,0,sizeof(m_debug));
@@ -733,6 +742,9 @@ omx_vdec::omx_vdec(): m_error_propogated(false),
 #else
     m_disable_ubwc_mode = true;
 #endif
+#else //_ANDROID_
+    char *env_ptr = getenv("OMX_DEBUG_LEVEL");
+    debug_level = env_ptr ? atoi(env_ptr) : 0;
 #endif
     memset(&m_cmp,0,sizeof(m_cmp));
     memset(&m_cb,0,sizeof(m_cb));
@@ -899,10 +911,12 @@ omx_vdec::~omx_vdec()
     pthread_mutex_destroy(&m_lock);
     pthread_mutex_destroy(&c_lock);
     sem_destroy(&m_cmd_lock);
+#ifdef _ANDROID_
     if (perf_flag) {
         DEBUG_PRINT_HIGH("--> TOTAL PROCESSING TIME");
         dec_time.end();
     }
+#endif
     DEBUG_PRINT_INFO("Exit OMX vdec Destructor: fd=%d",drv_ctx.video_driver_fd);
     m_perf_control.send_hint_to_mpctl(false);
 }
@@ -2182,11 +2196,15 @@ OMX_ERRORTYPE omx_vdec::component_init(OMX_STRING role)
             }
         }
 
+#ifdef _ANDROID_
         property_get("persist.vidc.dec.conceal_color", property_value, DEFAULT_CONCEAL_COLOR);
         m_conceal_color= atoi(property_value);
         DEBUG_PRINT_HIGH("trying to set 0x%u as conceal color\n", (unsigned int)m_conceal_color);
-        control.id = V4L2_CID_MPEG_VIDC_VIDEO_CONCEAL_COLOR;
         control.value = m_conceal_color;
+#else
+        control.value = atoi(DEFAULT_CONCEAL_COLOR);
+#endif
+        control.id = V4L2_CID_MPEG_VIDC_VIDEO_CONCEAL_COLOR;
         ret = ioctl(drv_ctx.video_driver_fd, VIDIOC_S_CTRL, &control);
         if (ret) {
             DEBUG_PRINT_ERROR("Failed to set conceal color %d\n", ret);
@@ -2756,7 +2774,11 @@ OMX_ERRORTYPE  omx_vdec::send_command_proxy(OMX_IN OMX_HANDLETYPE hComp,
 #endif
         if (cmd == OMX_CommandFlush && (param1 == OMX_CORE_INPUT_PORT_INDEX ||
                     param1 == OMX_ALL)) {
+#ifdef _ANDROID_
             if (android_atomic_add(0, &m_queued_codec_config_count) > 0) {
+#else
+            if (g_atomic_int_add(&m_queued_codec_config_count, 0) > 0) {
+#endif
                struct timespec ts;
 
                clock_gettime(CLOCK_REALTIME, &ts);
@@ -4401,6 +4423,7 @@ OMX_ERRORTYPE  omx_vdec::set_parameter(OMX_IN OMX_HANDLETYPE     hComp,
                                      break;
 
                                  }
+#ifdef _ANDROID_
         case OMX_QcomIndexParamVideoMetaBufferMode:
         {
             StoreMetaDataInBuffersParams *metabuffer =
@@ -4443,6 +4466,7 @@ OMX_ERRORTYPE  omx_vdec::set_parameter(OMX_IN OMX_HANDLETYPE     hComp,
                 }
                 break;
         }
+#endif
         case OMX_QcomIndexParamVideoDownScalar:
         {
             QOMX_INDEXDOWNSCALAR* pParam = (QOMX_INDEXDOWNSCALAR*)paramData;
@@ -5093,9 +5117,11 @@ OMX_ERRORTYPE  omx_vdec::get_extension_index(OMX_IN OMX_HANDLETYPE      hComp,
         *indexType = (OMX_INDEXTYPE)OMX_GoogleAndroidIndexGetAndroidNativeBufferUsage;
     }
 #endif
+#ifdef _ANDROID_
     else if (extn_equals(paramName, "OMX.google.android.index.storeMetaDataInBuffers")) {
         *indexType = (OMX_INDEXTYPE)OMX_QcomIndexParamVideoMetaBufferMode;
     }
+#endif
 #ifdef ADAPTIVE_PLAYBACK_SUPPORTED
     else if (extn_equals(paramName, "OMX.google.android.index.prepareForAdaptivePlayback")) {
         *indexType = (OMX_INDEXTYPE)OMX_QcomIndexParamVideoAdaptivePlaybackMode;
@@ -5255,7 +5281,9 @@ OMX_ERRORTYPE  omx_vdec::use_output_buffer(
     unsigned                         i= 0; // Temporary counter
     struct vdec_setbuffer_cmd setbuffers;
     OMX_PTR privateAppData = NULL;
+#ifdef _ANDROID_
     private_handle_t *handle = NULL;
+#endif
     OMX_U8 *buff = buffer;
     struct v4l2_buffer buf;
     struct v4l2_plane plane[VIDEO_MAX_PLANES];
@@ -5286,6 +5314,7 @@ OMX_ERRORTYPE  omx_vdec::use_output_buffer(
     if (eRet != OMX_ErrorNone)
        return eRet;
 
+#ifdef _ANDROID_
     if (dynamic_buf_mode) {
         *bufferHdr = (m_out_mem_ptr + i );
         (*bufferHdr)->pBuffer = NULL;
@@ -5310,6 +5339,7 @@ OMX_ERRORTYPE  omx_vdec::use_output_buffer(
         (*bufferHdr)->nAllocLen = sizeof(struct VideoDecoderOutputMetaData);
         return eRet;
     }
+#endif
 
     if (eRet == OMX_ErrorNone) {
 #if defined(_ANDROID_HONEYCOMB_) || defined(_ANDROID_ICS_)
@@ -5517,12 +5547,16 @@ OMX_ERRORTYPE  omx_vdec::use_output_buffer(
         }
 
         (*bufferHdr)->nAllocLen = drv_ctx.op_buf.buffer_size;
+#ifdef _ANDROID_
         if (m_enable_android_native_buffers) {
             DEBUG_PRINT_LOW("setting pBuffer to private_handle_t %p", handle);
             (*bufferHdr)->pBuffer = (OMX_U8 *)handle;
         } else {
             (*bufferHdr)->pBuffer = buff;
         }
+#else
+        (*bufferHdr)->pBuffer = buff;
+#endif
         (*bufferHdr)->pAppPrivate = privateAppData;
         BITMASK_SET(&m_out_bm_count,i);
     }
@@ -6656,6 +6690,7 @@ OMX_ERRORTYPE  omx_vdec::empty_this_buffer(OMX_IN OMX_HANDLETYPE         hComp,
         return OMX_ErrorBadPortIndex;
     }
 
+#ifdef _ANDROID_
     if (perf_flag) {
         if (!latency) {
             dec_time.stop();
@@ -6663,6 +6698,7 @@ OMX_ERRORTYPE  omx_vdec::empty_this_buffer(OMX_IN OMX_HANDLETYPE         hComp,
             dec_time.start();
         }
     }
+#endif
 
     if (arbitrary_bytes) {
         nBufferIndex = buffer - m_inp_heap_ptr;
@@ -6773,6 +6809,7 @@ OMX_ERRORTYPE  omx_vdec::empty_this_buffer_proxy(OMX_IN OMX_HANDLETYPE  hComp,
     }
     /* If its first frame, H264 codec and reject is true, then parse the nal
        and get the profile. Based on this, reject the clip playback */
+#ifdef _ANDROID_
     if (first_frame == 0 && codec_type_parse == CODEC_TYPE_H264 &&
             m_reject_avc_1080p_mp) {
         first_frame = 1;
@@ -6789,6 +6826,7 @@ OMX_ERRORTYPE  omx_vdec::empty_this_buffer_proxy(OMX_IN OMX_HANDLETYPE  hComp,
             return OMX_ErrorNone;
         }
     }
+#endif
 
     DEBUG_PRINT_LOW("ETBProxy: bufhdr = %p, bufhdr->pBuffer = %p", buffer, buffer->pBuffer);
     /*for use buffer we need to memcpy the data*/
@@ -6902,7 +6940,11 @@ if (buffer->nFlags & QOMX_VIDEO_BUFFERFLAG_EOSEQ) {
 
     if (buffer->nFlags & OMX_BUFFERFLAG_CODECCONFIG) {
         DEBUG_PRINT_LOW("Increment codec_config buffer counter");
+#ifdef _ANDROID_
         android_atomic_inc(&m_queued_codec_config_count);
+#else
+        g_atomic_int_inc(&m_queued_codec_config_count);
+#endif
     }
 
     rc = ioctl(drv_ctx.video_driver_fd, VIDIOC_QBUF, &buf);
@@ -6963,6 +7005,7 @@ OMX_ERRORTYPE  omx_vdec::fill_this_buffer(OMX_IN OMX_HANDLETYPE  hComp,
         OMX_IN OMX_BUFFERHEADERTYPE* buffer)
 {
     unsigned nPortIndex = 0;
+#ifdef _ANDROID_
     if (dynamic_buf_mode) {
         private_handle_t *handle = NULL;
         struct VideoDecoderOutputMetaData *meta;
@@ -7007,6 +7050,7 @@ OMX_ERRORTYPE  omx_vdec::fill_this_buffer(OMX_IN OMX_HANDLETYPE  hComp,
         buffer->nFilledLen = 0;
         buffer->nAllocLen = handle->size;
     }
+#endif
 
 
     if (m_state == OMX_StateInvalid) {
@@ -7088,12 +7132,14 @@ OMX_ERRORTYPE  omx_vdec::fill_this_buffer_proxy(
         return OMX_ErrorNone;
     }
 
+#ifdef _ANDROID_
     if (dynamic_buf_mode) {
         drv_ctx.ptr_outputbuffer[nPortIndex].offset = 0;
         drv_ctx.ptr_outputbuffer[nPortIndex].buffer_len = buffer->nAllocLen;
         buf_ref_add(nPortIndex);
         drv_ctx.ptr_outputbuffer[nPortIndex].mmaped_size = buffer->nAllocLen;
     }
+#endif
 
     pending_output_buffers++;
 #ifdef _COLOR_CONVERTION_
@@ -7721,6 +7767,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
         buffer->nFlags &= ~OMX_BUFFERFLAG_DATACORRUPT;
     }
 
+#ifdef _ANDROID_
     if (m_debug_extradata) {
         if (buffer->nFlags & QOMX_VIDEO_BUFFERFLAG_EOSEQ) {
             DEBUG_PRINT_HIGH("***************************************************");
@@ -7734,6 +7781,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
             DEBUG_PRINT_HIGH("***************************************************");
         }
     }
+#endif
 
 
     DEBUG_PRINT_LOW("fill_buffer_done: bufhdr = %p, bufhdr->pBuffer = %p, flags: 0x%x, timestamp: %lld",
@@ -7808,6 +7856,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
         if (buffer->nFilledLen > 0) {
             time_stamp_dts.get_next_timestamp(buffer,
                     is_interlaced && is_duplicate_ts_valid);
+#ifdef _ANDROID_
             if (m_debug_timestamp) {
                 {
                     OMX_TICKS expected_ts = 0;
@@ -7823,6 +7872,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
                     }
                 }
             }
+#endif
         }
     }
 
@@ -7833,6 +7883,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
             else
                 set_frame_rate(buffer->nTimeStamp);
 
+#ifdef _ANDROID_
             if (perf_flag) {
                 if (!proc_frms) {
                     dec_time.stop();
@@ -7852,6 +7903,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
                     proc_frms = 0;
                 }
             }
+#endif
         }
         if (buffer->nFlags & OMX_BUFFERFLAG_EOS) {
             prev_ts = LLONG_MAX;
@@ -7901,15 +7953,18 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
             }
         }
 
+#ifdef _ANDROID_
         // add current framerate to gralloc meta data
         if (m_enable_android_native_buffers && m_out_mem_ptr) {
             OMX_U32 buf_index = buffer - m_out_mem_ptr;
             setMetaData((private_handle_t *)native_buffer[buf_index].privatehandle,
                          UPDATE_REFRESH_RATE, (void*)&current_framerate);
         }
+#endif
 
         if (il_buffer) {
             log_output_buffers(il_buffer);
+#ifdef _ANDROID_
             if (dynamic_buf_mode) {
                 unsigned int nPortIndex = 0;
 #ifdef _COLOR_CONVERTION_
@@ -7936,6 +7991,7 @@ OMX_ERRORTYPE omx_vdec::fill_buffer_done(OMX_HANDLETYPE hComp,
                     return OMX_ErrorBadParameter;
                 }
             }
+#endif
             m_cb.FillBufferDone (hComp,m_app_data,il_buffer);
         } else {
             DEBUG_PRINT_ERROR("Invalid buffer address from get_il_buf_hdr");
@@ -8103,9 +8159,15 @@ int omx_vdec::async_message_process (void *context, void* message)
             if (omxhdr->nFlags & OMX_BUFFERFLAG_CODECCONFIG) {
 
                 DEBUG_PRINT_LOW("Decrement codec_config buffer counter");
+#ifdef _ANDROID_
                 android_atomic_dec(&omx->m_queued_codec_config_count);
                 if ((android_atomic_add(0, &omx->m_queued_codec_config_count) == 0) &&
                     BITMASK_PRESENT(&omx->m_flags, OMX_COMPONENT_FLUSH_DEFERRED)) {
+#else
+                g_atomic_int_dec_and_test(&omx->m_queued_codec_config_count);
+                if ((g_atomic_int_add(&omx->m_queued_codec_config_count, 0) == 0) &&
+                    BITMASK_PRESENT(&omx->m_flags, OMX_COMPONENT_FLUSH_DEFERRED)) {
+#endif
                     DEBUG_PRINT_LOW("sem post for CODEC CONFIG buffer");
                     sem_post(&omx->m_safe_flush);
                 }
@@ -9611,6 +9673,7 @@ OMX_ERRORTYPE omx_vdec::allocate_output_headers()
             return OMX_ErrorInsufficientResources;
         }
 #endif
+#ifdef _ANDROID_
         if (dynamic_buf_mode) {
             out_dynamic_list = (struct dynamic_buf_list *) \
                 calloc (sizeof(struct dynamic_buf_list), drv_ctx.op_buf.actualcount);
@@ -9619,6 +9682,7 @@ OMX_ERRORTYPE omx_vdec::allocate_output_headers()
                   out_dynamic_list[i].dup_fd = -1;
             }
         }
+#endif
 
         if (m_out_mem_ptr && pPtr && drv_ctx.ptr_outputbuffer
                 && drv_ctx.ptr_respbuffer) {
@@ -9926,6 +9990,7 @@ void omx_vdec::handle_extradata(OMX_BUFFERHEADERTYPE *p_buf_hdr)
                         }
                     }
 
+#ifdef _ANDROID_
                     if (m_enable_android_native_buffers) {
                         DEBUG_PRINT_LOW("setMetaData INTERLACED format:%d color_format: %x enable:%d mbaff:%d",
                                          payload->format, interlace_color_format ,enable,
@@ -9939,6 +10004,7 @@ void omx_vdec::handle_extradata(OMX_BUFFERHEADERTYPE *p_buf_hdr)
                                LINEAR_FORMAT, (void*)&interlace_color_format);
                         }
                     }
+#endif
                     if (client_extradata & OMX_INTERLACE_EXTRADATA) {
                         append_interlace_extradata(p_extra, payload->format,
                                       p_buf_hdr->nFlags & QOMX_VIDEO_BUFFERFLAG_MBAFF);
@@ -10221,8 +10287,13 @@ OMX_U32 omx_vdec::count_MB_in_extradata(OMX_OTHER_EXTRADATATYPE *extra)
 
 void omx_vdec::print_debug_extradata(OMX_OTHER_EXTRADATATYPE *extra)
 {
+#ifdef _ANDROID_
     if (!m_debug_extradata || !extra)
         return;
+#else
+    if (!extra)
+        return;
+#endif
 
 
     DEBUG_PRINT_HIGH(
@@ -10781,7 +10852,12 @@ void omx_vdec::request_perf_level(enum vidc_perf_level perf_level)
     struct v4l2_control control;
     char property_value[PROPERTY_VALUE_MAX] = {0};
 
+#ifdef _ANDROID_
     property_get("vidc.debug.turbo", property_value, "0");
+#else
+    property_value[0] = '1';
+    property_value[1] = '\0';
+#endif
     memset(&control, 0, sizeof(v4l2_control));
     control.id = V4L2_CID_MPEG_VIDC_SET_PERF_LEVEL;
     switch (perf_level) {
@@ -11444,10 +11520,16 @@ bool omx_vdec::perf_control::load_lib()
     if (m_perf_lib)
         return true;
 
+#ifdef _ANDROID_
     if((property_get("ro.vendor.extension_library", perf_lib_path, NULL) <= 0)) {
         DEBUG_PRINT_ERROR("vendor library not set in ro.vendor.extension_library");
         goto handle_err;
     }
+#else
+    size_t len = strlen("libqti-perfd-client.so");
+    memcpy(perf_lib_path, "libqti-perfd-client.so", len);
+    perf_lib_path[len] = '\0';
+#endif
 
     if ((m_perf_lib = dlopen(perf_lib_path, RTLD_NOW)) == NULL) {
         DEBUG_PRINT_ERROR("Failed to open %s : %s",perf_lib_path, dlerror());
-- 
1.9.1

